### DISCUSSION TITLE
How to use matplotlib to draw to an existing Axes object at C++ side?

### DISCUSSION THREAD

**Discussion by UUNagato** [pybind/pybind11#3470] [Q&A] [answered]
In my project, I want to call matplotlib to draw (mainly add patches) to an existing Axes object (created from python side) at C++ side. I am wondering how to do that with pybind11?

In other words, I create an Axes object in python. This Axes object is later used as a parameter to a C++ function. And the C++ function draws something (mainly add patches) to this Axes.

I don't know how to define a C++ function that takes an Axes object as the parameter and calls matplotlib to draw to it.

Thank you for the help.

**Comment by EricCousineau-TRI** [comment, marked as answer]
Have you had a chance to review the docs about accessing Python object via C++ API?
https://pybind11.readthedocs.io/en/stable/advanced/pycpp/object.html

**Reply by UUNagato** [reply]
@EricCousineau-TRI Thank you. I have solved my problem. I once believed using py::module_::import and other similar functions requires an embedded intepreter but I was wrong. May I ask another question? How can I check if a py::object parameter is a matplotlib Axes object?

**Reply by EricCousineau-TRI** [reply]
Glad it helped.

To check, you can either use `py::isinstance(obj, type)` ([docs](https://pybind11.readthedocs.io/en/stable/reference.html#_CPPv410isinstance6handle6handle)), or you can use `py::module_::import("builtins").attr("isinstance")` directly.
